debezium-core/src/main/java/io/debezium/relational/history/KafkaDatabaseHistory.java

```diff
@@ -505,11 +506,23 @@ public void initializeStorage() {
 
         try (AdminClient admin = AdminClient.create(this.producerConfig.asProperties())) {
 
-            // Find default replication factor
-            final short replicationFactor = getDefaultTopicReplicationFactor(admin);
+            NewTopic topic;
+            try {
+                // Create topic with optional replication factor to get broker default on Kafka 2.4+
```

Comment:
Could you add to the comment the reasoning for doing that approach with the fallback to the other API.